### PR TITLE
Feature/RC SEAT

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
### This implementation is based on [RPC design refactoring](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0168-rpc-design-refactoring.md), so this PR should be merged after https://github.com/smartdevicelink/sdl_core/pull/2250

Technical task : #1860
This PR is **ready** for review.

This pull request contains the implementation of new remote control module - SEAT
(proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0105-remote-control-seat.md)
For testing use the following test set:
https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/feature/rc_seat/test_sets/rc_SEAT.txt

Pull request with ATF scripts: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1910

For manual testing use:
HMI from : smartdevicelink/sdl_hmi#52
mobile application SPT from HockeyApp 20180213-Nightly-Android

### Risk
This PR makes **[major]** API changes.
### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc [hunk 1]
```diff
@@ -63,18 +63,19 @@ bool CheckIfModuleTypeExistInCapabilities(
...
-    return false;
+  const std::map<std::string, std::string> params = {
+      {enums_value::kRadio, strings::kradioControlCapabilities},
+      {enums_value::kClimate, strings::kclimateControlCapabilities},
+      {enums_value::kSeat, strings::kseatControlCapabilities}};
+  bool is_module_type_valid = false;
+  for (const auto& param : params) {
+    if (param.first == module_type) {
+      if (rc_capabilities.keyExists(param.second)) {
+        is_module_type_valid = true;
```
Comment (yang1070) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc]: add a `break;` after set is_module_type_valid = true?

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc [hunk 2]
```diff
@@ -145,30 +173,38 @@ bool CheckIfModuleDataExistInCapabilities(
...
+
+bool isModuleTypeAndDataMatch(const std::string& module_type,
+                              const smart_objects::SmartObject& module_data) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::map<std::string, std::string> data_mapping =
+      GetModuleTypeToDataMapping();
+  bool module_type_and_data_match = false;
+  for (const auto& data : data_mapping) {
+    if (data.first == module_type) {
+      module_type_and_data_match = module_data.keyExists(data.second);
```
Comment (yang1070) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: break after it is found

File: src/components/interfaces/MOBILE_API.xml [hunk 3]
```diff
@@ -1224,6 +1224,9 @@
         </element>
         <element name="MOBILE_APP">
         </element>
+        <element name="RADIO_TUNER">
```
Comment (yang1070) [inline src/components/interfaces/MOBILE_API.xml]: this does not relate to SEAT, please remove from this PR

File: src/appMain/hmi_capabilities.json [hunk 4]
```diff
@@ -469,6 +469,25 @@
                         "signalStrengthAvailable": true,
                         "stateAvailable": true
                     }
+                ],
+				"seatControlCapabilities": [
```
Comment (yang1070) [inline src/appMain/hmi_capabilities.json]: missing moduleName
"moduleName" : "driver seat" 
or "moduleName" : "DRIVER"

**Comment by LitvinenkoIra** [comment]
@yang1070 Fixed all comments c92e0f61f18a2790f181177ad7971950330bfe92

**Comment by yang1070** [comment]
@theresalech Ford has reviewed, tested and approved this PR, It is ready for Livio to review.

**Comment by yang1070** [comment]
@theresalech @LuxoftAKutsan 
This pull request contains the implementation of new remote control module - SEAT.
Ford approved this PR.
It shall be rebased to merge into develop branch after RPC refactoring work.

**Comment by LitvinenkoIra** [comment]
change base branch from feature/rc_refactoring (not actual branch) to feature/vehicle_info_plugin  (branch that contains all changes regarding PRC refactoring )

**Comment by LitvinenkoIra** [comment]
@JackLivio feature/rc_seat rebased on latest develop

**Pull request merged**